ext/dom/php_dom.c

```diff
@@ -24,9 +24,12 @@
 #if HAVE_LIBXML && HAVE_DOM
 #include "ext/standard/php_rand.h"
 #include "php_dom.h"
-#include "dom_arginfo.h"
+#include "php_dom_arginfo.h"
 #include "dom_properties.h"
 #include "zend_interfaces.h"
+#if defined(LIBXML_XPATH_ENABLED)
+#include "xpath_arginfo.h"
+#endif
```

Comment:
I assume xpath is kept separately because the whole class is conditional?

Reply:
That's true. Although, I see now (I didn't have the energy to test it yesterday) that only an empty function entry array is left dangling when `LIBXML_XPATH_ENABLED` is not defined. So should I move `DOMXPath` to the rest of the stubs?

Reply:
Or maybe I could use `generateCodeWithConditions()` to generate the array definition if it's necessary.

Reply:
I'll merge it like this, but I'm happy to iterate on conditional class declaration in a followup PR.